File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        chars = {}
        for char in s:
            if char in chars:
                chars[char] += 1
            else:
                chars[char] = 1
        
        for char in t:
            if char in chars:
                chars[char] -= 1
            else:
                return False
        
        for value in chars.values():
            if value != 0:
                return False

        return True
        

File: 0242-valid-anagram/0242-valid-anagram.cpp
#include <map>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> letters = {};
        for (char c: s){
            letters[c]++;
        }

        for (char c: t){
            letters[c]--;
        }

        for (const auto& pair: letters){
            if (pair.second != 0){
                return false;
            }
        }

        return true;
    }
};